session.go

```diff
@@ -956,8 +1017,28 @@ type dirtyTableOperation struct {
 	row    []types.Datum
 }
 
+// Commit overrides the Transaction interface.
+func (st *TxnState) Commit(goCtx goctx.Context) error {
+	if st.fail != nil {
```

Comment:
Does  `st.fail != nil` only happen in autocommit = 1? then we don't commit this transaction, what about autocommit = 0, because we do not call TxnState.Commit in this situation, the stmtTxn can rollback or not commit, and then we do `COMIT` statement, so we still can not Commit(because we do not set st.fail to nil) the transaction?

Reply:
> Does st.fail != nil only happen in autocommit = 1?

No, `st.fail` has nothing to do with autocommit, `st.fail != nil` never happen, I bet.